crates/lsp/src/lib.rs

```diff
@@ -239,7 +239,7 @@ fn url_to_code_description(url: &Option<String>) -> Option<CodeDescription> {
 }
 
 impl<L: LSPLang> Backend<L> {
-  pub fn new(client: Client, rules: RuleCollection<L>) -> Self {
+  pub fn new(client: Client, rules: anyhow::Result<RuleCollection<L>>) -> Self {
```

Comment:
I have modified the LSP to use `std::result::Result` and avoid `anyhow::Result`. The `find_rules` function still creates an `anyhow::Result` but before we pass it in to the `Backend::new` constructor, we convert it into a `std::result::Result`.

Can you help me understand the cost associated with using the different Results from different libraries? I chose `anyhow` because it has some convenience functions and it is being used in other parts of the ast-grep rust project (though not in the LSP).